### DISCUSSION TITLE
installing monai in develop mode doesn't build with cuda

### DISCUSSION THREAD

**Discussion by sarbabi** [Project-MONAI/MONAI#4811] [General]
after installing pytorch in a virtual env and torch.cuda.is_available() is True.

$BUILD_MONAI=1 python setup.py develop

starts with 

(venv_monai) [saeed@isigpu2]$ BUILD_MONAI=1 python setup.py develop
setup.py with torch 1.12.0+cu116
BUILD_MONAI_CPP=True, **BUILD_MONAI_CUDA=False**, TORCH_VERSION=11200.

I checked in the setup.py, seems like the reason is $CUDA_HOME environment variable is empty, configuring address in CUDA_HOME, still returns another error: no such folder in $CUDA_HOME/bin/nvcc.

I can't have sudo access to install nvcc, but I guess it should not be necessary, Or I'm missing something?

Any help appreciated.
Best,
Saeed

**Comment by wyli** [comment]
perhaps the change mentioned in this ticket would enable forced install? https://github.com/Project-MONAI/MONAI/issues/4621

**Reply by sarbabi** [reply]
not really, in setup.py this line is:
`BUILD_CUDA = (CUDA_HOME is not None) if torch.cuda.is_available() else FORCE_CUDA`

so in my setup since torch.cuda.is_available() is True, gives value False to BUILD_CUDA (since my CUDA_HOME is None)

I tried to find what should I provide as CUDA_HOME and tried some addresses with no success, since it tries to find some file in $CUDA_HOME/bin/nvcc.

**Reply by wyli** [reply]
ok, nvcc is needed... I think you can install it without sudo, for example: `conda install cudatoolkit-dev -c conda-forge`

**Reply by sarbabi** [reply]
Thanks, could install nvcc without root access using the method in this link:
https://github.com/pyg-team/pytorch_geometric/issues/392
Best, Saeed
